pkg/lib/session/idpsession/provider.go

```diff
@@ -188,7 +188,9 @@ func (p *Provider) accessWithID(ctx context.Context, id string, accessEvent acce
 
 	s.AccessInfo.LastAccess = accessEvent
 	defer func() {
-		err = p.accessSideEffects(ctx, s, accessEvent)
+		if err == nil && s != nil {
+			err = p.accessSideEffects(ctx, s, accessEvent)
+		}
```

Comment:
Which error in sentry does this fix correspond to?

if `s` is nil, it should have return at line 186, isn't it?

Reply:
https://authgear.sentry.io/issues/6100188706/?project=4507492287512576&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&statsPeriod=14d&stream_index=1

> if s is nil, it should have return at line 186, isn't it?

Yes. So I don't know why nil pointer exception could be triggered here. However, the err == nil check is needed here, because the original implementation always overrides the err returned by `p.Store.Update`.